CMakeLists.txt

```diff
@@ -32,6 +32,7 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake")
 set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 set(CMAKE_ENABLE_EXPORTS TRUE)
+option(ENABLE_LINKTIME_OPTIMIZATION "Perform link time optimization" ON)
```

Comment:
In general, I think that the default and options in cmake scripts should work differently

## What it is now

| User preference     | IPO available | IPO not available    |
|---------------------|---------------|----------------------|
| None (default)      | ON            | OFF (w/ warning)     |
| Explicitly enabled  | ON            | **OFF (w/ warning)** |
| Explicitly disabled | OFF           | OFF                  |

## What it should be

| User preference     | IPO available | IPO not available |
|---------------------|---------------|-------------------|
| None (default)      | ON            | OFF (maybe w/out warning)  |
| Explicitly enabled  | ON            | **Error**         |
| Explicitly disabled | OFF           | OFF               |

That is, if user enabled it and we cannot deliver, then error out.

Reply:
@ganeshmurthy my $0.02US - I agree with @jiridanek that if the user has explicitly configured IPO and it isn't provided it should fail. That prevents me from inadvertently benchmarking IPO vs non-IPO and getting misleading results. 

Again, just my humble opinion.

Reply:
@jiridanek @kgiusti I have made the requested changes, please take a look. Thanks.

Reply:
I have to make a couple more changes to this PR, hang on.

Reply:
I'm not entirely sure that there is any work to allow LTO now that cmake supports it.

The only reason there is specific work in proton is that we supported it on versions of cmake too early to support it directly.

We're moving to remove any specific support and just to rely on the CMAKE_INTERPROCEDURAL_OPTIMIZATION cmake setting. The support for ENABLE_LINKTIME_OPTIMIZATION that is still there is purely for backwards  compatiblilty of build scripts.

Reply:
What I'm saying is that the only thing you should need to do in a recent version of cmake (proton now requries 3.16 or later) is to set CMAKE_INTERPROCEDURAL_OPTIMIZATION=On. In a build script all you need to do is add "-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=On" to the cmake line.